src/Notification.vala

```diff
@@ -113,4 +118,19 @@ public class Notifications.Notification : GLib.Object {
 
         return text;
     }
+
+    private Gdk.Pixbuf? read_image_data (Variant img) {
```

Comment:
This method can be static and I think it would be clearer to name it something like `image_data_variant_to_pixbuf`

Reply:
Oh, one more thing, I think I'd like to see a type check of the variant at the start of this method before it's deserialized just to check nothing has sent bogus data that might crash the notification server.

Reply:
Renamed that method and added a type check. Thanks for the review!